src/viam/app/data_client.py

```diff
@@ -174,15 +174,15 @@ async def tabular_data_by_filter(
         return data
 
     async def binary_data_by_filter(
-        self,
-        filter: Optional[Filter] = None,
-        dest: Optional[str] = None,
+        self, filter: Optional[Filter] = None, dest: Optional[str] = None, include_binary: bool = False
```

Comment:
is defaulting to `False` the right behavior here? I would've expected `True` to be a better default but will defer to @dannenberg on this one

Reply:
i think you'll wanna do some extra work to make this nice (ex add nice errors and a better bool name)... but defaulting to false is likely correct as you can only return one actual file. https://viaminc.slack.com/archives/C0307PGHR7E/p1692649912349079?thread_ts=1692649715.146639&cid=C0307PGHR7E

Reply:
I can totally see it going both ways. I only made it default to false because True would make the amount of data received of course much larger and noticeably slow down the execution of the function,  so I opted for the faster choice. Also @dannenberg the method itself will return as many files as fit into the filter. The 100 vs 1 limit thing only affects how many files are included in a single request. The method repeatedly makes requests over and over again until it gets the right number of files, regardless of the value of include_binary. So if a filter matches 7 files, calling this method will always return a list of those 7 files, they just may or may not actually contain the binary data of the file depending on the value of this boolean. Do you still think it should default to false?

Reply:
as i understand it (based on devin hilly's comment), you will not get any binary data unless you set the limit to 1. the code should be changed to reflect this.

Reply:
gotcha. thanks.